### PR TITLE
Python versions

### PR DESCRIPTION
**Pull request by dylanmcreynolds**
This PR updates the minimum version of python to 3.9. Additionally, it changes the gh action to test 3.9, 3.10 and 3.11.

Along the way, this converts the installation use `pyproject.toml`

### CODE REVIEW

**Comment by LAShemilt** [comment]
Love the change to pyproject.toml, much easier to manage. However tests are breaking due to `_version.py` being deleted.

I think all it takes is to remove the contents of the `pyscicat/__init__.py` file which is currently:
```
from ._version import get_versions

__version__ = get_versions()["version"]
del get_versions
```
I think this now can become a blank file.

File: pyscicat/_version.py
Comment (bpedersen2) [inline pyscicat/_version.py:1]: This should not get checked in at all, instead add it  to .gitignore.

File: setup.py
Comment (bpedersen2) [inline setup.py:1]: I would  keep a setup.py with just a setup() call.

**Review by bpedersen2 — CHANGES_REQUESTED** [review]
Generally fine and a good idea.

File: pyscicat/_version.py
Comment (dylanmcreynolds) [reply pyscicat/_version.py:1]: Thanks, fixed

File: setup.py
Comment (dylanmcreynolds) [reply setup.py:1]: Is that in case people are installing with old versions of pip?

Comment (bpedersen2) [reply setup.py:1]: Yes, not only pip, but see https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

**Comment by bpedersen2** [comment]
The test somehow believes we want `python 3.1`, which is wrong. I guess  the versions in the matrix need quoting, otherwise they are treated as float, giving that result.

**Comment by dylanmcreynolds** [comment]
That was painful. Apologies if I just blasted you with notification emails. But I got the build to run with these new changes.

**Pull request merged**